contribute.md

```diff
@@ -3,12 +3,251 @@ layout: page
 title: Contribute
 ---
 
-Each subsystem is separated into individual repositories within the linux-system-roles GitHub project. Just open a new issue against the appropriate subsystem’s issue tracker to report bugs or request enhancements at the subsytem listed in
+Each subsystem is separated into individual repositories within the
+[linux-system-roles](https://github.com/linux-system-roles) GitHub project.
+Just open a new issue against the appropriate subsystem's issue tracker to
+report bugs or request enhancements.  New subsystem requests or feedback can be
+provided to the project's landing page at
+[linux-system-roles.github.io](https://linux-system-roles.github.io) Pull
+requests welcome!
 
-[https://github.com/linux-system-roles](https://github.com/linux-system-roles)
 
-New subsystem requests or feedback can be provided to the project’s landing page at linux-system-roles.github.io Pull requests welcome!
+## Code structure
 
-The subsystems are licensed under different licences. By submitting a patch for inclusion, you are agreeing to license your changes under the subsystem licence.
+The repository is structured as follows:
 
-For discussions and questions, please use our mailing list (archives) or join our IRC channel #systemroles on freenode.
+- `./defaults/` - Contains the default role configuration.
+
+- `./examples/` - Contains YAML examples for different configurations.
+
+- `./meta/` - Metadata of the project.
+
+- `./tasks/` - Declaration of the different tasks that the role is going to execute.
+
+- `./tests/playbooks/` - Contains the complete tests for the role. `./tests/test_*.yml`
+  are shims to run tests once for every provider. `./tests/tasks/` contains task
+  snippets that are used in multiple tests to avoid having the same code repeated 
+  multiple times.
+
+The rest of files in the root folder mostly serve as configuration files for diferent
+testing tools and bots that help with the manteinance of the project.
+
+
+The code files will always have the imports on the first place, followed by constants
+and in the last place, classes and methods. The style of python coding for this project
+is [**PEP 8**](https://www.python.org/dev/peps/pep-0008/), with automatic formatting
+thanks to [Python Black](https://black.readthedocs.io/en/stable/). Make sure to install
+the formatter, it will help you a lot throughout the whole coding process!
```

Comment:
Since we use `tox` for this, you don't really have to install anything, just run `tox -e black`, `tox -e flake8`, etc.  or just run `tox` by itself to run all configured tests.